sbi/inference/posteriors/base_posterior.py

```diff
@@ -50,8 +55,10 @@ def __init__(
         neural_net: nn.Module,
         prior,
         x_shape: torch.Size,
+        sample_with: str,
```

Comment:
I see your point, it would definitely work. My only concern is that inexperienced users might prefer setting `sample_with='mcmc'` (they might not know what slice sampling is). I think I'd like to keep it as it is for now, just to show explicitly that e.g. `slice` is a `mcmc` method, not a `rejection sampling` method.

I think in the long run, we might want something like this:
```python
from sbi.samplers import MCMC, Rejection, VariationalInference

inference = SNLE(prior)
_ = inference.append_simulations(theta, x).train()
sampler = MCMC(inference, method="slice_np")  # can only sample().
posterior = inference.build_posterior(sampler)  # can sample(), log_prob(), map()
```

But this would of course be a huge refactor.

Reply:
OK I agree, let's leave like this for now. 
As an intermediate step in a future PR we could also use `mcmc_slice`, `mcmc_hmc` etc, but this could be confusing as well.